src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13076,7 +13076,9 @@ LambdaExpressionSyntax parseLambdaExpressionWorker()
                     // Unparenthesized lambda case
                     // x => ...
                     // x!! => ...
-                    var identifier = this.ParseIdentifierToken();
+                    var identifier = (this.CurrentToken.Kind != SyntaxKind.IdentifierToken && this.PeekToken(1).Kind == SyntaxKind.EqualsGreaterThanToken) ?
```

Comment:
Could you clarify why we need to peek for an `=>` here? It seems like we already scanned and determined that this is a lambda?

Reply:
At this point, we may have `x => ...` or just `=> ...` (missing parameter name).